code/game/gamemodes/abandoned/abandoned.dm

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Patrick Meade
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//------------------------------------------------------------------------------
+
+/datum/game_mode
+	var/mob/claimed_the_prize = null
+	var/obj/structure/closet/hiding_place = null
+	var/obj/item/stack/spacecash/c1000000/money = null
+
+/datum/game_mode/abandoned
+	name = "Abandoned Station"
+	config_tag = "abandoned-station"
+	required_players = 0
+	required_enemies = 0
+	recommended_enemies = 0
+	votable = 0
+
+/datum/game_mode/abandoned/announce()
+	to_chat(world, "<B>The current game mode is - Abandoned Station!</B>")
+	to_chat(world, "<B>Rumors of a vast sum of cash left behind have drawn you to the [station_name()].</B>")
+	to_chat(world, "Search the lockers on the station to find the prize, and return to the arrivals shuttle.")
+	to_chat(world, "<B>Beware!</B> Other treasure hunters are also after the prize and may be hostile!")
+	to_chat(world, "Other dangers may also lurk on the Abandoned Station...")
+
+/datum/game_mode/abandoned/can_start()
+	return ..()
+
+/datum/game_mode/abandoned/pre_setup()
+	// pick a random closet
+	hiding_place = pick(get_station_closets())
+	// create a stack of 1,000,000 credits
+	money = new()
+	// put the money in the closet
+	hiding_place.contents += money
+	// log and message the name and location of the closet containing the money
+	log_and_message_admins("[hiding_place] at ([hiding_place.x],[hiding_place.y],[hiding_place.z]) contains the cash!")
+	// tell the caller we successfully completed pre-setup
+	return TRUE
+
+/datum/game_mode/abandoned/post_setup()
+	return ..()
+
+/datum/game_mode/abandoned/proc/get_station_closets()
+	var/list/obj/structure/closet/L = list()
+	for(var/obj/structure/closet/i in GLOB.closets)
+		if(is_station_level(i.z))
+			L += i
+	return L
+
+/datum/game_mode/abandoned/declare_completion()
+	// announce if somebody won or not
+	if(claimed_the_prize)
+		to_chat(world, "[claimed_the_prize] has escaped the abandoned station with the cash!")
+	else
+		to_chat(world, "Nobody recovered the cash from the abandoned station!")
+	// allow our parent to complete processing
+	return ..()
+
+/datum/game_mode/proc/auto_declare_completion_abandoned()
+	return TRUE
+
+/datum/game_mode/abandoned/check_finished()
+	// for each player in the game
+	for(var/mob/P in GLOB.player_list)
```

Comment:
This includes ghosts.
Taken from elsewhere in the codebase: `for(var/mob/dead/observer/G in GLOB.player_list)`

So, a `mob/P` could be a `mob/dead/observer` and exist in `GLOB.player_list`.
`mob/P` will also include aliens and simple animals. And cyborgs.
So, if a ghost a is standing at that place, they'll be considered as being `on_shuttle`.
Granted, a ghost shouldn't have any contents...or would they? That's a good question.

Suggestion:

```
for(var/mob/P in GLOB.player_list)
   if(P.stat == DEAD || !iscarbon(P))
       return FALSE
   <rest of code>
```